src/ln/channelmanager.rs

```diff
@@ -1233,6 +1233,9 @@ impl ChannelManager {
 
 	/// Call this upon creation of a funding transaction for the given channel.
 	///
+	/// Note that ALL inputs in the transaction pointed to by funding_txo MUST spend SegWit outputs
+	/// or your counterparty can steal your funds!
```

Comment:
Hmm may we prevent that with a call to ChainWatchInterface to be sure that inputs are all is_p2wsh/is_p2wpkh true ?

Reply:
Sadly we're not watching it and registering it may be expensive. The other alternative is to require the transaction be provided to funding_generated instead of just the txid and index. Dunno if its worth worrying about too much, though, as long as its well-documented.

Reply:
Okay, even if would prefer trigger guard better than documentation but may add surety later, if user didn't get it

Reply:
We can change funding_generated to panic if the user forgets to do it, certainly, but is definitely orthogonal to the rest of the patchset, I just slipped in the comments cause I realized we were missing them.